### DISCUSSION TITLE
enable to open file requested by external program

### DISCUSSION THREAD

**Discussion by 7flash** [helix-editor/helix#7799] [General]
I have external program window (obsidian excalidraw canvas) where I explore my files. Currently I have to copy file path and then enter ":open" command to see its content in my helix editor window. Instead I would like just to click on the file in my window and then see its content in my helix editor window.

Solution:
I can create click handler which writes the file path of chosen file into specific file on the disk. Then helix editor can watch changes in that specific file and trigger ":open" command with the path written in file.

**Comment by gabydd** [comment]
this is not currently possible directly in helix see: https://github.com/helix-editor/helix/issues/2177

but it's possible to send text to things like tmux/zellij/kitty and there are some examples floating around in different issues of how people have configured that if you look up one of those applications in the discussions and issues there should be some examples

**Reply by 7flash** [reply]
I am actually running helix inside of zellij multiplexer inside of alacritty terminal; should I then make some kind of plugin on the side of zellij which will run "hx /filepath" command in terminal? but could it rather somehow trigger ":open" command inside of running helix?

**Reply by 7flash** [reply]
going to try suggested approach, thanks for referencing: https://github.com/helix-editor/helix/issues/2177#issuecomment-1519182725

**Comment by 7flash** [comment]
Following my solution with running proxy server in deno over kitty: https://github.com/helix-editor/helix/issues/2177#issuecomment-1661237854
